packages/editor/src/components/post-schedule/index.js

```diff
@@ -2,14 +2,68 @@
  * WordPress dependencies
  */
 import { __experimentalGetSettings } from '@wordpress/date';
-import { withSelect, withDispatch } from '@wordpress/data';
-import { compose } from '@wordpress/compose';
+import { useDispatch, useSelect } from '@wordpress/data';
 import { DateTimePicker } from '@wordpress/components';
-import { useRef } from '@wordpress/element';
+import { useRef, useState, useMemo } from '@wordpress/element';
+import { store as coreStore } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import { store as editorStore } from '../../store';
+
+function getDayOfTheMonth( date = new Date(), firstDay = true ) {
+	const d = new Date( date );
+	return new Date(
+		d.getFullYear(),
+		d.getMonth() + ( firstDay ? 0 : 1 ),
+		firstDay ? 1 : 0
+	).toISOString();
+}
+
+export default function PostSchedule() {
+	const { postDate, postType } = useSelect(
+		( select ) => ( {
+			postDate: select( editorStore ).getEditedPostAttribute( 'date' ),
+			postType: select( editorStore ).getCurrentPostType(),
+		} ),
+		[]
+	);
+
+	const { editPost } = useDispatch( editorStore );
+	const onUpdateDate = ( date ) => editPost( { date } );
+
+	const [ previewedMonth, setPreviewedMonth ] = useState(
+		getDayOfTheMonth( postDate )
+	);
+
+	// Pick up published and schduled site posts.
+	const eventsByPostType = useSelect(
+		( select ) =>
+			select( coreStore ).getEntityRecords( 'postType', postType, {
+				status: 'publish,future',
+				after: getDayOfTheMonth( previewedMonth ),
+				before: getDayOfTheMonth( previewedMonth, false ),
+				exclude: [ select( editorStore ).getCurrentPostId() ],
+			} ),
+		[ previewedMonth, postType ]
+	);
+
+	const events = useMemo(
+		() =>
+			( eventsByPostType || [] ).map(
+				( { title, type, date: eventDate } ) => ( {
+					title: title?.raw,
```

Comment:
Do folks know if we need the raw or rendered value 🤔

Reply:
I wasn't sure either and since it isn't going to be consumed on this approach, I preferred using the raw version of the title. However, if we'd like to implement the events tooltip we may like to go for the rendered version to format the event list:

![image](https://user-images.githubusercontent.com/77539/113918394-c207cd80-97b8-11eb-9530-689ca614e282.png)

Reply:
PR https://github.com/WordPress/gutenberg/pull/29737

Reply:
Seems like this should be the rendered title, the raw is used when editing only AFAIK